### ISSUE TITLE
Multi-GPU Inference Mode

### ISSUE THREAD

**Issue by CMCDragonkai** [keras-team/keras#11844]
I'm testing out the multi GPU mode during inference. I have found that you need a batch size to be always divisible by the number of GPUs available. This poses a problem when a request asks to just infer for just 1 image and you have 2 GPUs (and therefore 2 set as the batch size in various configuration parameters). When asking to infer on a batch size of just 1, you end up with this error:

```
2018-12-11 05:41:28.297597: F tensorflow/stream_executor/cuda/cuda_dnn.cc:444] could not convert BatchDescriptor {count: 0 feature_map_count: 64 spatial: 112 112  value_min: 0.000000 value_max: 0.000000 layout: BatchDepthYX} to cudnn tensor descriptor: CUDNN_STATUS_BAD_PARAM
Aborted
```

I've tested using `predict`, `predict_on_batch` and `predict_generator` and they all result in the same issue if the batch size is not divisible by the number of GPUs. For 2 GPUs, they must end up with a batch size that is a multiple of the number of GPUs, 2 or 4.. .etc.

Since the training works even when the batch size is not divisible by the GPU, because the last batch size may be uneven with all batches as it is specifically mentioned in the Keras documentation: 

> Different batches may have different sizes. For example, the last batch of the epoch is commonly smaller than the others, if the size of the dataset is not divisible by the batch size.

Then shouldn't the inference mode be also capable of using multi GPUs with batch sizes that are not divisible by the number of GPUs? I'm not entirely sure how training even deals with uneven batch sizes when there are multiple GPUs, does 1 GPU end up doing nothing and just idling? Or can there be a round-robing scheduler, or task stealing scheduler?

**Comment by CMCDragonkai** [comment]
I should mention that I'm using Keras 2.1.4. I'm not sure if anything recently changed in Keras 2.2 to allow this to work: https://github.com/keras-team/keras/blob/2.2.4/keras/utils/multi_gpu_utils.py

**Comment by gabrieldemarmiesse** [comment]
When reporting bugs, please test first on the latest version of keras (not 2.2.4, keras built from the master branch of github).

**Comment by CMCDragonkai** [comment]
When I last checked the code, it didn't look like there was anything that would change the situation.

**Comment by alkamid** [comment]
Below is a MWE replicating this issue:
```python
from keras.layers import Input, Conv3D, MaxPool3D, Activation
from keras.models import Model
from keras.utils import multi_gpu_model
import numpy as np

inputs = Input(shape=(32, 32, 32, 1))
x = Conv3D(16, (3, 3, 3))(inputs)                                                                                                                                                                                                                                               
x = MaxPool3D(pool_size=(2, 2, 2))(x)
predictions = Activation('sigmoid')(x)

model = Model(inputs=inputs, outputs=predictions)
parallel_model = multi_gpu_model(model, gpus=2)

inp = np.random.random((3, 32, 32, 32, 1))                                                                                                                                                                                                                                      
pred = parallel_model.predict(inp, batch_size=2)
```

Running this on a multi GPU machine, I got the following error:
```
tensorflow/stream_executor/cuda/cuda_dnn.cc:503] Check failed: cudnnSetTensorNdDescriptor(handle_.get(), elem_type, nd, dims.data(), strides.data()) == CUDNN_STATUS_SUCCESS (3 vs. 0)batch_descriptor: {count: 0 feature_map_count: 16 spatial: 30 30 30  value_min: 0.000000 value_max: 0.000000 layout: BatchDepthYX}
Aborted (core dumped)
```

With the input divisible by batch_size (e.g. `inp = np.random.random((2, 32, 32, 32, 1))`), the scripts works fine. `MaxPool3D` layer seems to be the culprit --- this code works fine with that line removed or replaced by `x = Dense(16)(x)`. Analogous 2D model also works fine:
```python
inputs = Input(shape(32, 32, 1))
x = Conv2D(16, (3, 3))(inputs)
x = MaxPool2D(pool_size=(2, 2))(x)
predictions = ACtivateion('sigmoid')(x)
```

@CMCDragonkai, does your model comprise `MaxPool3D` layers? If not, could you try and find which layer(s) in your model is causing the problem?
